### DISCUSSION TITLE
Autoincrement using mongoosjs

### DISCUSSION THREAD

**Discussion by basavarajdodamani** [strapi/strapi#7834] [Legacy Topics]
Hello,
Could we implement autoincrement on a field of a collection using mongoosjs, just like sequence/trigger feature provided by usual RDBMS?
From a few of the articles, i found that it is not an OOTB feature:
https://github.com/strapi/strapi/issues/2784
https://github.com/strapi/strapi/issues/861

As there was not much of response to the similar implementation provided in first article, i just would like to know if this is possible? Or i could go with another option of maintaining a counter but which would not be an efficient way of doing it.

Kind Regards,
Basu

**Comment by derrickmehaffy** [comment]
We don't currently have any suggestions on the topic, pending the feature request: https://github.com/strapi/strapi/issues/2784 that you linked. 

For the moment the best possible solution would probably be adding an integer to the model and handling the auto-increment via the model lifecycles: https://strapi.io/documentation/v3.x/concepts/models.html#lifecycle-hooks

In where the lifecycle checks the database for the highest value then increases by one before creating the entry.
